### PR TITLE
 CosmosDB package 2015-04 update

### PR DESCRIPTION
**Pull request by yaohaizh**

### CODE REVIEW

**Comment by yaohaizh** [comment]
Related issue: https://github.com/Azure/autorest.java/issues/375

File: .azure-pipelines/fluentJavaMgmt.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ trigger:
   - master
 
 variables:
-  DefaultOptions: '--batch-mode -Dmaven.wagon.http.pool=false'
+  DefaultOptions: '-Dmaven.wagon.http.pool=false'
```
Comment (praries880) [inline .azure-pipelines/fluentJavaMgmt.yml]: Why did we need this change?

File: azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/ExtendedResourceProperties.java [hunk 2]
```diff
@@ -18,80 +18,80 @@
     /**
      * A system generated property. A unique identifier.
      */
-    @JsonProperty(value = "rid")
-    private String rid;
+    @JsonProperty(value = "_rid")
```
Comment (praries880) [inline azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/ExtendedResourceProperties.java]: @anuchandy can you look at what caused this change?

File: azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/implementation/GremlinDatabaseInner.java [hunk 3]
```diff
@@ -28,22 +28,22 @@
     /**
      * A system generated property. A unique identifier.
      */
-    @JsonProperty(value = "properties.rid")
-    private String rid;
+    @JsonProperty(value = "properties._rid")
```
Comment (praries880) [inline azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/implementation/GremlinDatabaseInner.java]: @anuchandy same as above here... why did this naming change?

File: .azure-pipelines/fluentJavaMgmt.yml (hunk 1 above)
Comment (yaohaizh) [reply .azure-pipelines/fluentJavaMgmt.yml]: If with batch mode, the maven job will always run from the parent pom directory. In this case, the suppressions.xml of cosmosdb will not work.  Thus, have to exclude batch mode in this case to run check style into each individual directory.

File: azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/ExtendedResourceProperties.java (hunk 2 above)
Comment (yaohaizh) [reply azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/ExtendedResourceProperties.java]: As I checked the cosmos-db.json, these properties are always start with `-`, not sure how the value is `rid` in our code for json property.

Which way should we honor here?

Comment (anuchandy) [reply azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/ExtendedResourceProperties.java]: The swagger indeed has underscore as part of property names https://github.com/Azure/azure-rest-api-specs/blob/99d1cd23e2058369ea1d0bba1ad042f69503a18e/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json#L3935

**Pull request merged**